src/main/java/gyro/google/compute/RegionInstanceGroupManagerResource.java

```diff
@@ -0,0 +1,193 @@
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Immutable;
+import gyro.core.resource.Resource;
+import gyro.core.scope.State;
+import gyro.core.validation.Required;
+
+/**
+ * Creates a region instance group manager.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     google::compute-region-instance-group-manager region-instance-group-manager-example
+ *         name: "region-instance-group-manager-example"
+ *         base-instance-name: "region-instance-group-manager-example"
+ *         description: "Region instance group manager example"
+ *         instance-template: $(google::compute-instance-template instance-group-template-example)
+ *         target-size: 1
+ *         region: "us-central"
+ *     end
+ */
+@Type("compute-region-instance-group-manager")
+public class RegionInstanceGroupManagerResource extends AbstractInstanceGroupManagerResource {
+
+    private ComputeDistributionPolicy distributionPolicy;
+
+    private String region;
+
+    /**
+     * Policy specifying intended distribution of instances in regional managed instance group.
+     *
+     * @subresource gyro.google.compute.ComputeDistributionPolicy
+     */
+    public ComputeDistributionPolicy getDistributionPolicy() {
+        return distributionPolicy;
+    }
+
+    public void setDistributionPolicy(ComputeDistributionPolicy distributionPolicy) {
+        this.distributionPolicy = distributionPolicy;
+    }
+
+    /**
+     * The region where the managed instance group resides.
+     */
+    @Immutable
+    @Required
+    public String getRegion() {
+        return region;
+    }
+
+    public void setRegion(String region) {
+        this.region = region;
+    }
+
+    @Override
+    public boolean doRefresh() throws Exception {
+        Compute client = createClient(Compute.class);
+        copyFrom(client.regionInstanceGroupManagers().get(getProjectId(), getRegion(), getName()).execute());
+        return true;
+    }
+
+    @Override
+    public void doCreate(GyroUI ui, State state) throws Exception {
+        InstanceGroupManager instanceGroupManager = createInstanceGroupManager();
+        Optional.ofNullable(getDistributionPolicy())
+            .map(ComputeDistributionPolicy::copyTo)
+            .ifPresent(instanceGroupManager::setDistributionPolicy);
+        Compute client = createClient(Compute.class);
+        Operation operation = client.regionInstanceGroupManagers()
+            .insert(getProjectId(), getRegion(), instanceGroupManager)
+            .execute();
+        waitForCompletion(client, operation);
+        refresh();
+    }
+
+    @Override
+    public void doUpdate(GyroUI ui, State state, Resource current, Set<String> changedFieldNames)
+        throws Exception {
+        // TODO: target pool resource
+        // https://github.com/perfectsense/gyro-google-provider/issues/79
+        //    private List<AbstractTargetPoolResource> targetPools;
```

Comment:
Should be removed and be done as a separate PR when target pool is available

Reply:
Yeah, it should be handled in a separate PR, but better to leave the comment as it will be helpful when implementing it.